src/source/spatial.rs

```diff
@@ -39,12 +45,9 @@ where
     pub fn set_positions(
         &mut self, emitter_pos: [f32; 3], left_ear: [f32; 3], right_ear: [f32; 3],
     ) {
-        let emitter_position = Point3::from(emitter_pos);
-        let left_ear = Point3::from(left_ear);
-        let right_ear = Point3::from(right_ear);
-        let left_distance = (left_ear - emitter_position).magnitude();
-        let right_distance = (right_ear - emitter_position).magnitude();
-        let max_diff = (left_ear - right_ear).magnitude();
+        let left_distance = dist(left_ear, emitter_pos);
+        let right_distance = dist(right_ear, emitter_pos);
+        let max_diff = dist(left_ear, right_ear);
         let left_diff_modifier = ((left_distance - right_distance) / max_diff + 1.0) / 4.0 + 0.5;
         let right_diff_modifier = ((right_distance - left_distance) / max_diff + 1.0) / 4.0 + 0.5;
         let left_dist_modifier = (1.0 / left_distance.powi(2)).min(1.0);
```

Comment:
And adjusting these `left_dist_modifier` and right_dist_modifier` lines.

Reply:
Good idea, thanks!